subgenerator/index.js

```diff
@@ -1,25 +1,25 @@
 'use strict';
 var path = require('path');
-var generators = require('yeoman-generator');
+var Generator = require('yeoman-generator');
 var superb = require('superb');
 
-module.exports = generators.Base.extend({
-  constructor: function () {
-    generators.Base.apply(this, arguments);
+module.exports = class extends Generator {
+  constructor(args, opts) {
+    super(args, opts);
 
-    this.argument('namespace', {
+    this.option('name', {
```

Comment:
Mhm. Using `this.argument('name', {` leads to failing test.

Tests are failing with:
```
  1) generator:app defaults "before all" hook:
     Error: Did not provide required argument name!
      at .<anonymous> (node_modules/yeoman-generator/lib/index.js:335:33)
      at Array.forEach (native)
      at Base.checkRequiredArgs (node_modules/yeoman-generator/lib/index.js:331:19)
      at Base.parseOptions (node_modules/yeoman-generator/lib/index.js:316:8)
      at argument (node_modules/yeoman-generator/lib/index.js:246:8)
      at new <anonymous> (subgenerator/index.js:9:447)
      at Environment.instantiate (node_modules/yeoman-environment/lib/environment.js:336:10)
      at composeWith (node_modules/yeoman-generator/lib/index.js:485:26)
      at default (app/index.js:9:2522)
      at Object.<anonymous> (node_modules/yeoman-generator/lib/index.js:387:23)

  2) generator:subgenerator "before all" hook:
     Uncaught Error: Did not provide required argument name!
      at .<anonymous> (node_modules/yeoman-generator/lib/index.js:335:33)
      at Array.forEach (native)
      at Base.checkRequiredArgs (node_modules/yeoman-generator/lib/index.js:331:19)
      at Base.parseOptions (node_modules/yeoman-generator/lib/index.js:316:8)
      at argument (node_modules/yeoman-generator/lib/index.js:246:8)
      at new <anonymous> (subgenerator/index.js:9:447)
      at Environment.instantiate (node_modules/yeoman-environment/lib/environment.js:336:10)
      at Environment.create (node_modules/yeoman-environment/lib/environment.js:313:15)
      at RunContext._run (node_modules/yeoman-test/lib/run-context.js:90:29)
      at RunContext.<anonymous> (node_modules/yeoman-test/lib/run-context.js:58:10)
```